sql/xsql/src/main/scala/org/apache/spark/sql/xsql/XSQLExternalCatalog.scala

```diff
@@ -594,7 +594,9 @@ private[xsql] class XSQLExternalCatalog(conf: SparkConf, hadoopConf: Configurati
 
   override def setCurrentDatabase(dbName: String): Unit = {
     withWorkingDSDB(dbName) { (_, db, _, _) =>
-      currentDataBase = Option(db)
+      synchronized{
```

Comment:
Why there need synchronized ?

Reply:
I have checked previous implement. 
```
def setCurrentDatabase(dsName: String, dbName: String): Unit = {
      setWorkingDataSource(Option(dsName)) {
        super.setCurrentDatabase(dbName)
        externalCatalog.setCurrentDatabase(dbName)
      }
}
```
This is the only entrance. The tread safe has provided by setWorkingDataSource, so we not need this change.